ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt

```diff
@@ -41,4 +44,33 @@ class PaywallActivityLauncher {
     fun launch(offering: Offering? = null) {
         activityResultLauncher.launch(PaywallActivityArgs(offeringId = offering?.identifier))
     }
+
+    /**
+     * Launch the paywall activity if the current user does not have [requiredEntitlementIdentifier] active.
+     * @param offering The offering to be shown in the paywall. If null, the current offering will be shown.
+     */
+    @JvmOverloads
+    suspend fun launchIfNeeded(
```

Comment:
Not sure if this is the best way to do this (exposing a `suspend` method). Feedback welcome.

Reply:
Hmm well, I imagine many people using activities/fragments are still not using coroutines, so offering a wrapper would be better... However, we would only be able to launch this coroutine from a global scope unless we ask developers to give us the launcher activity/fragment/view model to act as the scope...

I see 2 options here:
- Store a weak reference to the activity/fragment on the constructor so we can use that to launch the coroutine in that scope, and offer a non-coroutine version of these methods.
- Have a non-coroutine version of `shouldDisplayPaywall` that returns a callback so we can support this use case. We can wrap that callback in a coroutine for the current version, same way we do in `CoroutinesExtensionsCommon`.

Of those 2, I think the second might be the cleanest, but the first one is probably ok.
Any other ideas @vegaro ?

Reply:
Thanks! Just did option 2 :)